### PR TITLE
libzip: update to 1.7.3

### PR DESCRIPTION
**Pull request by Biswa96**

### CODE REVIEW

File: mingw-w64-libzip/0001-lib-compat-use-snprintf.patch [hunk 1]
```diff
@@ -0,0 +1,11 @@
+--- libzip-1.7.3.orig/lib/compat.h
++++ libzip-1.7.3/lib/compat.h
+@@ -94,7 +94,7 @@
+ #if !defined(HAVE_FILENO) && defined(HAVE__FILENO)
+ #define fileno _fileno
+ #endif
+-#if defined(HAVE__SNPRINTF)
++#if !defined(HAVE__SNPRINTF)
```
Comment (mati865) [inline mingw-w64-libzip/0001-lib-compat-use-snprintf.patch:8]: This seems very wrong, you cannot use `_snprintf` if it doesn't exist.

Comment (Biswa96) [reply mingw-w64-libzip/0001-lib-compat-use-snprintf.patch:8]: " if it doesn't exist" -- example? The build and test went fine.

Comment (mati865) [reply mingw-w64-libzip/0001-lib-compat-use-snprintf.patch:8]: CMake detects if `_snprintf` exists here: https://github.com/nih-at/libzip/blob/c55cf12cec53a5ef519f98dfd11f4cb00e9ac0f1/CMakeLists.txt#L95
This patch defines `snprintf` as `_snprintf` only when it doesn't exist which is wrong.

If you want to workaround GCC bug mentioned in https://github.com/msys2/MINGW-packages/issues/6736#issuecomment-666154781 it'd be better to disable the check.

Comment (Biswa96) [reply mingw-w64-libzip/0001-lib-compat-use-snprintf.patch]: Done. Thank you ❤️

**Pull request merged**